File: tests/tester.java
// This file is written assuming the AlphaKey class is imported.
public class Tester {
  public static void main(String[] args) {
    AlphaKey myAlphaKey = new AlphaKey();
    myAlphaKey.setTestingLength(3); // Will start testing from three character strings.
    myAlphaKey.setDebugFunction(setDebugFunction(new AlphaKey.DebugTester() {
        @Override
        public void Debug(String result, String guess, int index) {
            System.out.println("TEST INFO: "+index+": "+result+" ["+guess+"]");
        }
    })); // Print testing info
    AlphaKey.Tester myTester = new AlphaKey.Tester(){
      @Override
      public String Test(String guess) {
        if(guess.equals("jes")){
          return "Jessica";
        }
        return "JESSICA";
      }
    };
    myAlphaKey.testAgainst(myTester,"Jessica");
  }
}

File: tests/tester.py
import sys,pprint,imp #Import Modules
AlphaKey = imp.load_source('module.name', '../lib/AlphaKey.py') #Import alphaKey
def debug_function(result,guess,index):
    print(str(index)+": "+guess)
myAlphaKey = AlphaKey.AlphaKey({'IL':"JS",'debug_function':debug_function}) #Create an alphaKey object
myAlphaKey.setTestingMaxValue(100)
myAlphaKey.setTestingLength(7)
myAlphaKey.testAgainst(lambda guess:0,"jes")
